docs/commit-messages.md

```diff
@@ -108,6 +108,46 @@ For changes in `tailscale/tailscale` that fix a significant bug or add a new fea
 add `RELNOTE: <summary of change>` toward the end of the commit message.
 This will aid the release engineer in writing the release notes for the next release.
 
+## Is it a #cleanup?
+
+Our issuebot permits writing `Updates #cleanup` instead of an actual GitHub issue number.
+
+But only do that if it’s actually a cleanup. Don’t use that as a shortcut to avoid filing a bug. Shortcuts to file bugs: http://go/bugc (corp, safe choice) or http://go/bugo (open source, if you want it public to the world).
+
+The following guide can help you decide whether a tracking issue is warranted.
+
+|     |     |
+| --- | --- |
+| Was there a crash/panic? | Not a cleanup. Put the panic in a bug. Talk about when it was introduced, why, why a test didn’t catch it, note what followup work might need to be done. |
+| Did a customer report it? | Not a cleanup. Make a corp bug with links to the customer ticket. |
+| Is it from an incident, get paged? | Not a cleanup. Let’s track why we got paged. |
+| Does it change behavior? | Not a cleanup. File a bug to track why. |
+| Adding a test for a recently fixed bug? | Not a cleanup. Use the recently fixed bug’s bug number. |
+| Does it tweak a constant/parameter? | Not a cleanup. File a bug to track the debugging/tuning effort and record past results and goals for the future state. |
+| Fixing a regression from an earlier change? | Not a cleanup. At minimum, reference the PR that caused the regression, but if users noticed, it might warrant its own bug. |
+| Is it part of an overall effort that’ll take a hundred small steps? | Not a cleanup. The overall effort should have a tracking bug to collect all the minor efforts. |
+| Is it a security fix? Is it a security hardening? | Not a cleanup. There should be a bug about security incidents or security hardening efforts and backporting to previous releases, etc. |
+| Is it a feature flag being removed? | Not a cleanup. File a task to coordinate with other teams and to track the work. |
+
+### Actual cleanup examples
+
+- Fixing typos in internal comments that users would’ve never seen
+- Moving away from a deprecated API to an equivalently behaving newer API
```

Comment:
(unless it's part of series of a many commits, per line 128 above.... if it's a big migration effort, it should have a tracking bug)

Reply:
I think the previous wording covers too many use cases. How about the following rewording, with some examples:

```suggestion
- Simple, mechanical replacement of a deprecated API to its equivalently behaving replacement
  - [`errors.Wrapf`](https://pkg.go.dev/github.com/pkg/errors#Wrapf) → [`fmt.Errorf("%w")`](https://pkg.go.dev/fmt#Errorf)
  - [math/rand](https://pkg.go.dev/math/rand) → [math/rand/v2](https://pkg.go.dev/math/rand/v2)
```

Reply:
Reworded with the above suggestion in [682136d..e4d822d](https://github.com/tailscale/tailscale/compare/682136d6652de1f98ada85348418242818680b37..e4d822d1e916549590960308282cae0c51261e28).